### ISSUE TITLE
Php version not detected or working

### ISSUE THREAD

**Issue by mabasic** [phpactor/phpactor#2405]
I'm trying to setup phpactor to work with a php7.4 project, but it seems like it does not respect any php version set.

I have set the version in `.phpactor.yml`

```
php.version: 7.4
```

I have set the version in composer.json:

```
  "require": {
    "php": "^7.4",
    ...
  },
  "config": {
    "platform": {
      "php": "^7.4"
    }
  }
```

But whatever I do, when I place this code inside a php file, I get no warnings or errors, nothing:

```php
enum Suit
{
    case Hearts;
    case Diamonds;
    case Clubs;
    case Spades;
}

class Test {
    public readonly Bool $something;

    public function __construct(protected String $test) {
        $this->something = false;
    }

    public function getSomething(): String {
        $this->something = true;
        return $this->test;
    }
}

$test = new Test(test: "Mario");
echo $test->getSomething();

$test->something = true;
```

I expect to get some kind of an error like enum not supported, or invalid syntax, even the property promotion or whatever it is called, even the readonly does not work.

**Comment by dantleech** [comment]
Not sure what the issue is. Are you expecting Phpactor to show errors that you are using enums/etc in a 7.4 project?

**Comment by mabasic** [comment]
Yes

**Comment by mamazu** [comment]
The PHP version is currently not taken into account when analyzing source code. The syntax errors would be bound to the version of the analyzer package which in phpactor's case already supports 8.0 features.

What might make sense is to add a diagnostic to check for missing features. But that would be quite a big undertaking keeping up with what is allowed in which version.

**Comment by mabasic** [comment]
ah, I see. So, this is something that could be improved upon or implemented. I got this idea because in PHPStorm you get an error saying that you are using a feature that does not exist in the selected PHP version.

I have added instruction to the [Helix editor wiki](https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers#phpactor) on how to use Phpactor LSP with it. It works very well for me. Maybe we could add somewhere in the Phpactor documentation that Helix editor is supported also?
